safenode/src/protocol/wallet/keys.rs

```diff
@@ -19,7 +19,15 @@ const MAIN_KEY_FILENAME: &str = "main_key";
 /// Filename for storing the node's reward (BLS hex-encoded) public address.
 const PUBLIC_ADDRESS_FILENAME: &str = "public_address";
 
+/// Parse a public address from a hex-encoded string.
+#[allow(clippy::result_large_err)]
```

Comment:
Is there a reason the `Error` grew so big?

Reply:
It's something but didn't delve into it. Deferred.